src/sqlfluff/dialects/dialect_postgres.py

```diff
@@ -3004,6 +3018,7 @@ class InsertStatementSegment(BaseSegment):
         "INSERT",
         "INTO",
         Ref("TableReferenceSegment"),
+        Ref("InsertIntoAliasExpressionSegment", optional=True),
```

Comment:
Why not just this?:

```suggestion
        Sequence("AS", Ref("SingleIdentifierGrammar"), optional=True),
```

Reply:
I was going to just do this, but the motivation for using a separate segment with the `alias_expression` type is that I think rules that analyse tables and aliases rely on it being a separate segment with the  `alias_expression` type in order to detect it.

Reply:
And the existing `AliasExpressionSegment` segment was too permissive in this situation and led to parsing errors.

Reply:
Can you think of a better name than `InsertIntoAliasExpressionSegment` as that might help? 😅

Reply:
That makes sense, and see you did comment to that effect in the PR description but I missed it.

Fancy adding a little comment to that effect to the `InsertIntoAliasExpressionSegment ` class in case anyone else wonders why this was done to in the future?

Would be nice if Sequence took and optional type argument (I presume it doesn't?) but to be honest if typing something then that's probably reason enough make it a separate segment.

Reply:
`AsAliasExpressionSegment`? That would also allow it to be used elsewhere if needed.

Reply:
I've updated the segment name and description 😄